src/cclib/parser/daltonparser.py

```diff
@@ -737,20 +754,36 @@ def extract(self, inputfile, line):
         # @    Final gradient norm:           0.000003746706
         # ...
         #
+        if "Final HF energy" in line and not (hasattr(self, "mpenergies") or hasattr(self, "ccenergies")):
+            self.metadata["theory"] = "HF"
```

Comment:
This term is used to identify the computational method conducted in a calculation. It can be semiempirical methods such as "AM1", "PM3" , or multi-configuration methods such as  "CASPT2", "GVB", etc.  I don't think we need to go deep into its theoretical background.